src/bs_size.c

```diff
@@ -474,6 +482,11 @@ BSSize bs_size_new_from_str (const char *size_str, BSError **error) {
     }
 
     loc_size_str = replace_char_with_str (size_str, '.', radix_char);
+    if (!loc_size_str) {
+        set_error (error, BS_ERROR_INVALID_SPEC, strdup_printf ("Failed to parse size spec: %s", size_str));
```

Comment:
This is equally nasty, the `set_error()` function should've been made to take printf-style variable arguments.

Reply:
I agree, but I don't think it's worth rewriting it now. We can blame @vpodzime for not designing this better :-)

Reply:
Hey, I had to leave some work for you guys! :wink: